### ISSUE TITLE
Underlying SQL Errors Masked When Transaction Has Been Aborted

### ISSUE THREAD

**Issue by jturkel** [afair/postgresql_cursor#14]
To reproduce run the following:

```
ActiveRecord::Base.each_row_by_sql('select * from bad_table') { }
```

This leads to the following exception:

```
ActiveRecord::StatementInvalid: PG::InFailedSqlTransaction: ERROR:  current transaction is aborted, commands ignored until end of transaction block
: close cursor_1
```

It would be much easier to troubleshoot issues if the original exception was thrown:

```
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "bad_table" does not exist
LINE 1: declare cursor_1 cursor for select * from bad_table
                                                  ^
: declare cursor_1 cursor for select * from bad_table
```

**Comment by AndrewWhitaker** [comment]
Hi, I'm still running into this when statement timeouts occur inside of a cursor fetch. Why was @tijmenb's change not merged in? It fixes the problem for me.

**Comment by afair** [comment]
The fix was made in commit 6ffa646 (see above). Perhaps other work made the PR un-mergable?  I see the fix is different and may handle more cases (suppress errors in close). I'll look at that later.

**Comment by tijmenb** [comment]
I don't remember this specific instance, but I tend to close PRs if they've been open for a while so I'm guessing that's what happened here. If you want to reopen it that's fine by me.
